react/src/lib/components/WebvizDialog/WebvizDialog.tsx

```diff
@@ -0,0 +1,732 @@
+        ).map((elm) => {
+            return parseInt((elm as HTMLElement).style.zIndex);
+        });
+
+        if (placeholderDiv) {
+            const dialogZIndex = parseInt(placeholderDiv.style.zIndex);
+            const highestZIndex = Math.max(...zIndexes);
+            if (dialogZIndex !== highestZIndex) {
+                placeholderDiv.style.zIndex = `${highestZIndex}`;
+                placeholders.forEach((placeholder) => {
+                    const zIndex = parseInt(
+                        (placeholder as HTMLElement).style.zIndex
+                    );
+                    if (
+                        placeholder !== placeholderDiv &&
+                        zIndex > dialogZIndex
+                    ) {
+                        (placeholder as HTMLElement).style.zIndex = `${
+                            zIndex - 1
+                        }`;
+                    }
+                });
+            }
+        }
+    }, [dialogRef.current, placeholderDiv, props.modal, props.id]);
+
+    React.useLayoutEffect(() => {
+        if (!placeholderDiv) {
+            return;
+        }
+        setPlaceholderClassNameAndZIndex(placeholderDiv, props.modal || false);
+    }, [props.modal]);
+
+    React.useLayoutEffect(() => {
+        let node: HTMLDivElement | null = null;
+        if (!document.getElementById("WebvizDialog__root")) {
+            node = document.createElement("div");
+            node.setAttribute("id", "WebvizDialog__root");
+            document.body.appendChild(node);
+        } else {
+            node = document.getElementById(
+                "WebvizDialog__root"
+            ) as HTMLDivElement;
+        }
+
+        const placeholder = document.createElement("div");
+        setPlaceholderClassNameAndZIndex(placeholder, props.modal || false);
+        node.appendChild(placeholder);
+        setPlaceholderDiv(placeholder);
+
+        return () => {
+            if (placeholder) {
+                placeholder.parentElement?.removeChild(placeholder);
+            }
+            const dialogRoot = document.getElementById("WebvizDialog__root");
+            if (dialogRoot) {
+                const dialogNodes =
+                    dialogRoot.getElementsByClassName("WebvizDialog");
+                if (
+                    (dialogNodes.length == 1 &&
+                        dialogNodes[0] === dialogRef.current) ||
+                    dialogNodes.length === 0
+                ) {
+                    document.body.removeChild(dialogRoot);
+                }
+            }
+        };
+    }, []);
+
+    React.useLayoutEffect(() => {
+        if (props.open === open) {
+            return;
+        }
+
+        setOpen(props.open || false);
+        if (props.open) {
+            handleSetActive();
+        }
+    }, [props.open]);
+
+    const handleClose = React.useCallback(
+        (reason: DialogCloseReason) => {
+            if (reason == DialogCloseReason.BACKDROP_CLICK && !props.modal) {
```

Comment:
Better `===`. Is a backdrop click even possible when the dialog is not modal?

Reply:
Good point. No need for `reason` in component. Removed and deleted `DialogCloseReason`-enum